pkg/policy/selectorcache.go

```diff
@@ -1093,11 +1122,15 @@ func (sc *SelectorCache) RemoveIdentitiesFQDNSelectors(fqdnSels []api.FQDNSelect
 	sc.releaseIdentityMappings(identitiesToRelease)
 }
 
-// GetLabels must be called while holding sc.mutex!
-func (sc *SelectorCache) GetLabelsLocked(id identity.NumericIdentity) labels.LabelArray {
+// GetNetsLocked returns the most specific CIDR for an identity. For the "World" identity
+// it returns both IPv4 and IPv6.
+func (sc *SelectorCache) GetNetsLocked(id identity.NumericIdentity) []*net.IPNet {
 	ident, ok := sc.idCache[id]
 	if !ok {
-		return labels.LabelArray{}
+		return nil
+	}
+	if !ident.computed {
+		panic("identity not initialized properly")
```

Comment:
This makes me a bit nervous. I get why the `panic` from a developer's point of view, but worried that it gets hit in production. If that happens, I'm not sure what the mitigation would be besides inform the user to delete their policies. 

I think I'd rather turn this into an error log where we say "SelectorCache does not contain the expected pre-computed identity mapping. If a broad CIDR deny rule is applied, along with ToFQDNs allow rule which overlaps with the deny CIDR, then either rule may not be taken into account properly. Please report this to the developers.".

Reply:
Good call, changed to a warning log instead of a panic!